web/src/ResourceGroupsContext.tsx

```diff
@@ -67,10 +71,25 @@ export function ResourceGroupsContextProvider(
     function getGroup(groupLabel: string) {
       return groups[groupLabel] ?? { ...DEFAULT_GROUP_STATE }
     }
+
+    // We can expand all groups by resetting the collapse state to empty."
+    function expandAll() {
+      setGroups({}) // Reset state.
+    }
+
+    // To collapse all groups, we need to know all their names.
+    function collapseAll(groupNames: string[]) {
+      let newState: GroupsState = {}
+      groupNames.forEach((group) => (newState[group] = { expanded: false }))
+      setGroups(newState)
```

Comment:
this is a minor detail / edge case that i'm not sure is worth addressing, but i'll mention it anyways! when disabled resources are hidden, label groups with _only_ disabled resources will be reset to expanded when `collapseAll` is invoked.

Reply:
oh boy it gets more complicated than that!

- if you click "collapse all", and then create new groups, do they default to expanded or collapsed?
- if you remove a collapsed group, click "expand all", and then bring the group back, does it default to expanded or collapsed?

i don't think there's an obvious right answer here...i like how the existing model encourages a default state of "expanded" when there are any open questions.

will leave some comments on this.